application/cms/models.py

```diff
@@ -190,6 +190,34 @@ class DataSource(db.Model, CopyableModel):
         "MeasureVersion", secondary="data_source_in_measure_version", back_populates="data_sources"
     )
 
+    def merge(self, data_source_ids=[]):
+
+        if self.id in data_source_ids:
+            raise ValueError("Can’t merge with self")
+
+        if len(data_source_ids) == 0:
+            raise ValueError("No data source IDs specified")
+
+        data_sources_to_delete = DataSource.query.filter(DataSource.id.in_(data_source_ids))
+
+        if (data_sources_to_delete.count()) != len(data_source_ids):
+
+            difference = len(data_source_ids) - data_sources_to_delete.count()
+
+            raise ValueError(f"#{difference} data sources not found")
+
+        for data_source_to_delete in data_sources_to_delete:
+
+            DataSourceInMeasureVersion.query.filter(
+                DataSourceInMeasureVersion.data_source_id == data_source_to_delete.id
+            ).update({"data_source_id": self.id})
+
+            db.session.delete(data_source_to_delete)
+
+        db.session.commit()
+
+        return True
```

Comment:
I don't think we need to return anything here, really. This seems pretty arbitrary.